command/operator_migrate.go

```diff
@@ -385,19 +420,16 @@ func dfsScan(ctx context.Context, source physical.Backend, cb func(ctx context.C
 				}
 			}
 		} else {
-			err := cb(ctx, key)
-			if err != nil {
-				return err
-			}
+			// Pooling
+			eg.Go(func() error {
+				permitPool.Acquire()
+				defer permitPool.Release()
```

Comment:
For clarity, I am wondering why do we need to use two similar concurrent tools together? 
I don't think permitPool would have access to the max limit of the storage backend as well.

Reply:
Agree, thanks Hamid! I've removed the permit pool now.